components/wpa_supplicant/esp_supplicant/src/esp_wpa2.c

```diff
@@ -1193,3 +1195,63 @@ esp_err_t esp_wifi_sta_wpa2_use_default_cert_bundle(bool use_default_bundle)
     return ESP_FAIL;
 #endif
 }
+
+esp_err_t esp_wifi_sta_wpa2_ent_eap_noob_set_initial_association(void)
+{
+    g_wpa_eap_noob_state.active = true;
+    g_wpa_eap_noob_state.persistent = false;
+    g_wpa_eap_noob_state.noob_state = 0;
+    g_wpa_eap_noob_state.ephemeral_state = (struct eap_noob_ephemeral_state_info *)os_zalloc(sizeof(struct eap_noob_ephemeral_state_info));
+    g_wpa_eap_noob_state.supported_dir = EAP_NOOB_OOB_DIRECTION_SERVER_TO_PEER;
+    if (g_wpa_eap_noob_state.ephemeral_state == NULL) {
+        return ESP_ERR_NO_MEM;
+    }
+
+    return ESP_OK;
+}
+esp_err_t esp_wifi_sta_wpa2_ent_eap_noob_set_persistent_association(char *peer_id, int version, int cryptosuite, int cryptosuite_prev, char *nai, u8 *kz, u8 *kz_prev)
+{
+    return ESP_OK;
+}
+
+eap_noob_oob_msg_t *esp_wifi_sta_wpa2_ent_eap_noob_generate_oob_message(void)
+{
+    return eap_noob_generate_oob_msg();
+}
+
+esp_err_t esp_wifi_sta_wpa2_ent_eap_noob_receive_oob_message(u8 *noob, u8 *hoob)
+{
+    eap_noob_oob_msg_t *oobmsg = os_zalloc(sizeof(eap_noob_oob_msg_t));
+    memcpy(oobmsg->noob, noob, 16);
```

Comment:
I suggest using `sizeof(oobmsg->noob)` here to avoid the magic number.
(Same for hoob in the next line).